realm/realm-library/src/androidTest/kotlin/io/realm/MixedBulkInsertsTests.kt

```diff
@@ -0,0 +1,556 @@
+        }
+
+        val value1 = MixedNotIndexedWithPK(0)
+        value1.mixed = Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_3))
+
+        val value2 = MixedNotIndexedWithPK(1)
+        value2.mixed = Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_4))
+
+        realm.beginTransaction()
+        val objects = realm.copyToRealmOrUpdate(arrayListOf(value1, value2))
+        realm.commitTransaction()
+
+        val managedInnerObject1 = objects[0].mixed!!.asRealmModel(PrimaryKeyAsString::class.java)
+        val managedInnerObject2 = objects[1].mixed!!.asRealmModel(PrimaryKeyAsString::class.java)
+
+        assertTrue(managedInnerObject1.isManaged)
+        assertEquals(TEST_VALUE_3, managedInnerObject1.name)
+
+        assertTrue(managedInnerObject2.isManaged)
+        assertEquals(TEST_VALUE_4, managedInnerObject2.name)
+    }
+
+    @Test
+    fun bulk_copyToRealmOrUpdate_realmModelList() {
+        realm.executeTransaction { realm ->
+            val value1 = realm.createObject<MixedRealmListWithPK>(0)
+            value1.mixedList = RealmList(*testList1)
+            value1.mixedList.add(Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_1)))
+
+            val value2 = realm.createObject<MixedRealmListWithPK>(1)
+            value2.mixedList = RealmList(*testList1)
+            value2.mixedList.add(Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_1)))
+        }
+
+        val value1 = MixedRealmListWithPK(0)
+        value1.mixedList = RealmList(*testList2)
+        value1.mixedList.add(Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_3)))
+
+        val value2 = MixedRealmListWithPK(1)
+        value2.mixedList = RealmList(*testList2)
+        value2.mixedList.add(Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_4)))
+
+        realm.beginTransaction()
+        val objects = realm.copyToRealmOrUpdate(arrayListOf(value1, value2))
+        realm.commitTransaction()
+
+        assertTrue(objects[0].mixedList.containsAll(testList2.asList()))
+        assertTrue(objects[1].mixedList.containsAll(testList2.asList()))
+
+        val managedInnerObject1 = objects[0].mixedList.last()!!.asRealmModel(PrimaryKeyAsString::class.java)
+        val managedInnerObject2 = objects[1].mixedList.last()!!.asRealmModel(PrimaryKeyAsString::class.java)
+
+        assertTrue(managedInnerObject1.isManaged)
+        assertEquals(TEST_VALUE_3, managedInnerObject1.name)
+
+        assertTrue(managedInnerObject2.isManaged)
+        assertEquals(TEST_VALUE_4, managedInnerObject2.name)
+    }
+
+    @Test
+    fun bulk_insert_realmModel() {
+        realm.executeTransaction { realm ->
+            val value1 = MixedNotIndexedWithPK(0)
+            value1.mixed = Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_1))
+
+            val value2 = MixedNotIndexedWithPK(1)
+            value2.mixed = Mixed.valueOf(PrimaryKeyAsString(TEST_VALUE_2))
+
+            realm.insert(arrayListOf(value1, value2))
+        }
+
+        val objects = realm.where<MixedNotIndexedWithPK>().findAll()!!
```

Comment:
If you need `!!` here and in the other tests, something is wrong with our nullability annotations

Reply:
They were not needed!